src/picongpu/include/simulation_defines/param/gasConfig.param

```diff
@@ -22,219 +22,231 @@
+ *
+ * He (2e- / Atom ) with 1.e15 He / m^3
+ *                      = 2.e15 e- / m^3 */
+const double GAS_DENSITY_SI = 1.e25;
 
-    }
+}
 
-    //##########################################################################
-    //############## special gas profiles ######################################
-    //##########################################################################
+//##########################################################################
+//############## special gas profiles ######################################
+//##########################################################################
 
-    namespace gasGaussian
+namespace gasProfiles
+{
+
+struct GaussianParam
+{
+    /** Gas Formula:
+            const float_X exponent = fabs((y - GAS_CENTER) / GAS_SIGMA);
+            const float_X density = __expf(GAS_FACTOR*__powf(exponent, GAS_POWER));
+
+    takes GAS_CENTER_LEFT for y < GAS_CENTER_LEFT,
+          GAS_CENTER_RIGHT for y > GAS_CENTER_RIGHT,
+    and exponent = float_X(0.0) for GAS_CENTER_LEFT < y < GAS_CENTER_RIGHT */
+    static const float_X GAS_FACTOR = -1.0;
+    static const float_X GAS_POWER = 4.0;
+
+    /** height of vacuum area on top border
+     *      this vacuum is really important because of the laser initialization,
+     *      which is done in the first cell of the simulation
+     *  unit: cells */
+    static const uint32_t VACUUM_CELLS_Y = 50;
+
+    struct SI
     {
-        /** Gas Formula:
-                    const float_X exponent = fabs((y - GAS_CENTER) / GAS_SIGMA);
-                    const float_X density = __expf(GAS_FACTOR*__powf(exponent, GAS_POWER));
-
-            takes GAS_CENTER_LEFT for y < GAS_CENTER_LEFT,
-                  GAS_CENTER_RIGHT for y > GAS_CENTER_RIGHT,
-            and exponent = float_X(0.0) for GAS_CENTER_LEFT < y < GAS_CENTER_RIGHT */
-        const float_X GAS_FACTOR = -1.0;
-        const float_X GAS_POWER = 4.0;
-
-        namespace SI
-        {
-            /** The central position of the gas distribution
-             *  unit: meter */
-            const double GAS_CENTER_LEFT_SI = 4.62e-5;
-            const double GAS_CENTER_RIGHT_SI = 4.62e-5;
-            /** the distance from GAS_CENTER until the gas density decreases to its 1/e-th part
-             *  unit: meter */
-            const double GAS_SIGMA_LEFT_SI = 4.0e-5;
-            const double GAS_SIGMA_RIGHT_SI = 4.0e-5;
-        }
+        /** The central position of the gas distribution
+         *  unit: meter */
+        static const float_64 GAS_CENTER_LEFT = 4.62e-5;
+        static const float_64 GAS_CENTER_RIGHT = 4.62e-5;
+        /** the distance from GAS_CENTER until the gas density decreases to its 1/e-th part
+         *  unit: meter */
+        static const float_64 GAS_SIGMA_LEFT = 4.0e-5;
+        static const float_64 GAS_SIGMA_RIGHT = 4.0e-5;
+    };
+};
 
-    }
+/* definition of gas with gaussian profile*/
+typedef GaussianImpl<GaussianParam> Gaussian;
```

Comment:
as we discussed offline: this is not flexible enough and very obfuscating for the user.

a user want's to add several predefined objects together with self-written formulas to describe a structure or profile, e.g. a grid of spheres (profile=sum_over_x(profile(x)), a modulation of a self-defined profile (profile=p1*p2) or a "negative" imprint of an object in an other gas profile (profile=profile1-profile2) ...